.github/workflows/pr-changeset-review.yml

```diff
@@ -0,0 +1,35 @@
+name: "pr-changeset-review"
+
+# This workflow runs lint-like checks against changesets in PRs. The linting tool used is called Vale. The worflow will
+# add comments for any issues warnings or errors that Vale finds in the changeset.
+
+# To run vale locally against changesets, you can use `pnpm run check:changesets`.
+
+on:
+  pull_request:
+    types:
+      - opened # PR is created
+      - synchronize # commits added to PR
+      - reopened # closed PR re-opened
+    branches:
+      - main
+    paths:
+      - ".changeset/**" # Trigger only when changes are found under .changeset
+
+permissions:
+  pull-requests: write
+
+jobs:
+  vale:
+    name: vale
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # ratchet:actions/checkout@v3
+        with:
+          persist-credentials: false
+      - uses: errata-ai/vale-action@v2.1.1
```

Comment:
I'm not familiar with the repository for actions; do we need to pin to a hash here too? Or is it not possible for someone to publish different code under the same tag/version?

Reply:
good catch - I forgot to run ratchet on the file to pin.